packages/cli-kit/src/public/node/api/admin.ts

```diff
@@ -121,11 +136,18 @@ async function fetchApiVersions(session: AdminSession): Promise<ApiVersion[]> {
  *
  * @param store - Store FQDN.
  * @param version - API version.
+ * @param session - User session.
  * @returns - Admin API URL.
  */
-export function adminUrl(store: string, version: string | undefined): string {
+export function adminUrl(store: string, version: string | undefined, session?: AdminSession): string {
   const realVersion = version ?? 'unstable'
-  return `https://${store}/admin/api/${realVersion}/graphql.json`
+  const themeKitAccessDomain = process.env[environmentVariables.themeKitAccessDomain] ?? defaultThemeKitAccessDomain
+
+  const url =
+    session && isThemeAccessSession(session)
```

Comment:
Yes in the issue description I mentioned `In follow up, would like to clean up theme access logic by creating a file for it, rather than implementing it in REST and GraphQL`. I was thinking of doing it after since this work is blocking other CLI GraphQL PRs but I can try to do it here

Reply:
https://github.com/Shopify/cli/issues/4739